pkg/router/template/template_helper.go

```diff
@@ -157,28 +157,16 @@ func processEndpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit, action
 }
 
 func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
-	if len(alias.PreferPort) == 0 && len(svc.EndpointTable) != 1 {
-		// We can skip the work.  They are selecting everything, and since they don't have one endpoint,
-		// we can't disable the health checks.  So we can just use their list.
+	if len(alias.PreferPort) == 0 {
 		return svc.EndpointTable
 	}
-
 	endpoints := make([]Endpoint, 0, len(svc.EndpointTable))
-	for _, endpoint := range svc.EndpointTable {
-		// Filter the list:
-		// - If PreferPort length is 0, match everything
-		// - Otherwise, if the PortName or Port matches PreferPort, then we have a match
-		if len(alias.PreferPort) == 0 || endpoint.PortName == alias.PreferPort || endpoint.Port == alias.PreferPort {
+	for i := range svc.EndpointTable {
+		endpoint := svc.EndpointTable[i]
+		if endpoint.PortName == alias.PreferPort || endpoint.Port == alias.PreferPort {
```

Comment:
do you need the len(alias.PreferPort) == 0 to avoid a segfault?

Reply:
@pecameron 
No, len(alias.PreferPort) == 0 implies user did not specific any preferred port (http/https), then we don't need to filter any endpoints. That case is already handled in Line 160

Reply:
There are two commits in this PR. The first simply reverts my previous bad PR restoring the previous behavior. It probably makes sense to look at the second commit only.